CONTRIBUTING.md

```diff
@@ -47,6 +47,25 @@ The snapshots are checked into CI and are reviewed as part of the PR. They ensur
 
 [Read the docs](https://insta.rs/docs/) for more information on cargo insta.
 
+## `test.env`
+
+This file controls on how tests are executed.
+
+### Plan Tests
+
+- `TEST_PLAN_PIN_PKGS` - If set, it will pin the nixpkgs archive during testing.
+- `TEST_PLAN_CUSTOM_START_CMD` - Sets a custom start command.
+- `TEST_PLAN_CUSTOM_PKGS` - Sets custom nix packages.
+- `TEST_PLAN_ENVS` - Sets custom environment variables.
+
+### Docker Tests
+
+- `TEST_DOCKER_ENVS` - Sets custom environment variables.
+- `TEST_DOCKER_NO_PIN_PKGS` - If set, it will **not** pin the nixpkgs archive during testing.
+- `TEST_DOCKER_CUSTOM_PKGS` - Sets custom nix packages.
+- `TEST_DOCKER_EXPECTED_OUTPUT` - The expected output of the program that is being tested.
+- `TEST_DOCKER_WITH_POSTGRES` - If set, it will mount postgres to the test.
```

Comment:
Why not just use the same shared envs to configure nixpacks for the plan/build (e.g. envs, custom packages, etc)?